blocks/search-results/search-results.js

```diff
@@ -0,0 +1,42 @@
+import { createCardBlock } from '../card/card.js';
+import { loadBlock } from '../../scripts/lib-franklin.js';
+
+export async function searchResults(params, jsonData) {
+  // eslint-disable-next-line
+    return jsonData.data.filter((entry) => (entry.title + entry.content + entry.path + entry.authors + entry.collections + entry.section + entry.categories + entry.template).toLowerCase().includes(params.toLowerCase()));
+}
+
+export default async function decorate(block) {
+  // fetch results from json files
+  block.innerHTML = '';
+  console.log(window.location.origin);
+  const allData = await fetch(`${window.location.origin}/articles.json?sheet=full`);
+  console.log(allData);
+  const jsonData = await allData.json();
+  // for search functionality
+  const searchTerm = new URLSearchParams(window.location.search).get('q');
+  console.log(searchTerm);
+  block.classList.add('card-container', 'three-columns');
+  if (searchTerm) {
+    document.querySelector('.section.search-page-heading').innerHTML = `<h1>${searchTerm}</h1>`;
```

Comment:
Don't use unsanitized user input here with `innerHTML`. 
This simple method should clean it up enough: 
```js
/**
 * make text safe to use in innerHTML
 * @param text any string
 * @return {string} sanitized html string
 */
function plainText(text) {
  const fragment = document.createElement('div');
  fragment.append(text);
  return fragment.innerHTML;
}
```

https://issue28-searchfunctionality-v2--24life--hlxsites.hlx.live/search?q=OOPS%3Cimg%20src=https://miro.medium.com/v2/resize:fit:1400/0*y2OAF_DSarBAjihO.jpg%3E

<img width="1398" alt="Screenshot 2023-09-08 at 18 08 05" src="https://github.com/hlxsites/24life/assets/2250964/5b249c6a-1bc7-4152-a1fe-33b6db1002f0">